Helper/Config.php

```diff
@@ -157,48 +155,94 @@ public function getEnvironmentVariable(string $envVarName): ?string
     }
 
     /**
-     * Returns if the merchant is Active for NS8
+     * Returns if a given store is active for NS8
      *
+     * @param int|null $storeId The store ID
      * @return bool
      */
-    public function isMerchantActive(): bool
+    public function isMerchantActive(?int $storeId = null): bool
     {
-        return (bool) $this->scopeConfig->getValue(self::IS_MERCHANT_ACTIVE);
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->isActive : false;
     }
 
     /**
-     * Retrieve an access token.
-     *
-     * @return string|null The NS8 Protect Access Token.
+     * Retrieve the Protect access token for a given store.
+     * @param int|null $storeId ID of the store to retrieve token for
+     * @return string|null The NS8 Protect Access Token for the provided store.
+     */
+    public function getAccessToken(?int $storeId = null): ?string
+    {
+        $metadata = $this->getStoreMetadata($storeId);
+        return $metadata !== null ? $metadata->token : null;
+    }
+
+    /**
+     * Set the Protect access token for a given store.
+     * @param int|null $storeId The store ID to set token for
+     * @param string $token The Protect access token
+     * @return void
+     */
+    public function setAccessToken(?int $storeId, string $token): void
+    {
+        $metadata = new ProtectMetadata($token, $this->isMerchantActive($storeId));
+        $this->setStoreMetadata($storeId, $metadata);
+    }
+
+    /**
+     * Gets ProtectMetadata instance for a given store
+     * @param int|null $storeId The store ID to get metadata for
+     * @return ProtectMetadata
+     */
+    public function getStoreMetadata(?int $storeId): ?\stdClass
+    {
+        $metadatas = $this->getStoreMetadatas();
+        $storeId = $storeId !== null ? $storeId : self::EMPTY_STORE_ID;
+        return isset($metadatas[$storeId]) ? $metadatas[$storeId] : null;
+    }
+
+    /**
+     * Get all metadatas, for all stores.
+     * Structured as { [storeId: string]: ProtectMetadata };
+     * @return ProtectMetadata[]
      */
-    public function getAccessToken(): ?string
+    public function getStoreMetadatas(): array
     {
-        return $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $rawTokensJson = $this->encryptor->decrypt($this->scopeConfig->getValue(self::ACCESS_TOKEN_CONFIG_KEY));
+        $accessTokens = (array)json_decode($rawTokensJson);
+        return $accessTokens;
     }
 
     /**
-     * Saves a Magento configurable value in an encrypted format
+     * Stores a Protect metadata object for a given store.
      *
-     * @param string $key The key of the configurable value
-     * @param string $value The value we want to store and encrypt for the configurable value
+     * @param int|null $storeId ID of the store to store metadata for
+     * @param string $metadata Metadata object to store
```

Comment:
\@param ProtectMetadata $metadata...

Reply:
yep, originally I had these methods as just storing the access token, hence strings everywhere